src/auth/mysql.rs

```diff
@@ -0,0 +1,362 @@
+#![allow(unused_qualifications)]
+//! MySql authenticator module
+use argon2rs;
+
+use diesel::prelude::*;
+use diesel::mysql::MysqlConnection;
+use diesel::result::ConnectionError;
+
+// FIXME: Remove dependency on `ring`.
+use ring::test;
+use ring::constant_time::verify_slices_are_equal;
+
+use {Error, JsonMap, JsonValue};
+use super::{AuthenticationResult, Basic};
+
+/// Error mapping for `ConnectionError`
+impl From<ConnectionError> for Error {
+    fn from(e: ConnectionError) -> Error {
+        Error::GenericError(e.to_string())
+    }
+}
+
+// Code for conversion to hex stolen from rustc-serialize:
+// https://doc.rust-lang.org/rustc-serialize/src/rustc_serialize/hex.rs.html
+
+/// MySql user record
+#[derive(Queryable)]
+pub struct User {
+    username: String,
+    pw_hash: String,
+    salt: String,
+}
+
+/// A simple authenticator that uses a MySql backed user database.
+///
+/// Requires the `mysql_authenticator` feature, which is enabled by default.
```

Comment:
"which is enabled by default." should be removed. It is not enabled by default, and I think it shouldn't be.